internal/engine/upper.go

```diff
@@ -953,3 +956,8 @@ func handleTiltfileLogAction(ctx context.Context, state *store.EngineState, acti
 	state.CurrentTiltfileBuild.Log = model.AppendLog(state.CurrentTiltfileBuild.Log, action, state.LogTimestamps)
 	state.TiltfileCombinedLog = model.AppendLog(state.TiltfileCombinedLog, action, state.LogTimestamps)
 }
+
+func handleOpenWebUIAction(ctx context.Context, state *store.EngineState, action OpenWebUIAction) {
+	state.OpenedWebUIOnStart = true
+	_ = browser.OpenURL(action.WebURL.String())
```

Comment:
please do not use reducers this way. the really are supposed to be pure and not have side-effects.